lib/ash/actions/read/read.ex

```diff
@@ -3183,19 +3183,24 @@ defmodule Ash.Actions.Read do
          _context,
          load_attributes?
        ) do
-    if query.limit == 0 do
-      {:ok, []}
-    else
-      result
-      |> Helpers.select(query)
-      |> Helpers.load_runtime_types(query, load_attributes?)
-      |> case do
-        {:ok, result} ->
-          Ash.load(result, query, domain: query.domain, reuse_values?: true)
+    cond do
+      query.limit == 0 ->
+        {:ok, []}
 
-        other ->
-          other
-      end
+      query.filter == false ->
+        {:ok, []}
```

Comment:
Let's add a debug log here showing the resource and action name and saying that we skipped a query due to a filter being `false`